### ISSUE TITLE
Class Cast Exception - INFANTRY WEAPONS BRANCH. 

### ISSUE THREAD

**Issue by HammerGS** [MegaMek/megamek#3521]
**THIS IS AN ISSUE IN THE INFANTRY WEAPON BRANCH**

Working on adding the infantry weapons from Shrapnel to MegaMek. For testing loaded a Nighthawk XXX BA and equipped a Barton AMR. In the lobby the became marked as Invalid. In game when I go to use the weapon get the following error. 

```
megamek.MegaMek.lambda$0(MegaMek.java:68) - Uncaught Exception Detected
java.lang.ClassCastException: class megamek.common.MiscType cannot be cast to class megamek.common.AmmoType (megamek.common.MiscType and megamek.common.AmmoType are in unnamed module of loader 'app')
	at megamek.client.ui.swing.unitDisplay.WeaponPanel.setFieldofFire(WeaponPanel.java:1943)
	at megamek.client.ui.swing.unitDisplay.WeaponPanel.displaySelected(WeaponPanel.java:1920)
	at megamek.client.ui.swing.unitDisplay.WeaponPanel.valueChanged(WeaponPanel.java:2533)
```

### Environment

What version of MegaMek does your issue apply to?  MegaMek running from IntelliJ
What operating system are you using?  Windows
What java version are you using?  Java 11

**Comment by gsparks3** [comment]
I just reproduced this in version 49.7 using the Nighthawk XXX holding an Auto Grenade Launcher. I am fairly confident it is either related to or identical to #3494. Seems likely that it pops up for any BA holding an infantry weapon in its Armored Gloves _that the unit file was not constructed with_ (weapons assigned in the unit file seem to work fine).

**Comment by HammerGS** [comment]
Yeah, I'd bet that the issue is linked.

**Comment by gsparks3** [comment]
Further update - this issue applies to Anti-Personnel Weapon Mounts as well as Armored Gloves. Just tested in 49.7 with a Gray Death Scout Suit defined in the lobby as mounting a Mauser 960 rather than its standard Auto-Rifle. Again, the standard weapon defined in the unit file works normally.

**Comment by SJuliez** [comment]
This was apparently fixed with #4492
